### ISSUE TITLE
Fix an unexpected error with `as` Type Assertion in TSX

### ISSUE THREAD

**Issue by daku10** [markuplint/markuplint#1579]
- Markuplint version: `4.6.0`
- Parser lang: TSX
- Node.js version: `20.12.2`
- OS: ubuntu on WSL2 on windows

## Describe the bug

When I use `as` Type Assertion in attribute value, `invalid-attr` is thrown.
In my environment, `v4.5.0` has no error.

## Code Example

```tsx
import { useState } from "react";

type Value = "value1" | "value2";
export function App() {
  const [value, setValue] = useState<Value>("value1");

  return (
    <div>
      <select
        onChange={(e) => {
          setValue(e.target.value as Value);
        }}
      >
        <option>value1</option>
        <option>value2</option>
      </select>
      <p>{value}</p>
    </div>
  );
}
```

## Steps To Reproduce

1. clone [daku10/markuplint-issue-repo.git](https://github.com/daku10/markuplint-issue-repo)
2. npm install && npm run markuplint

## The current behavior

```
<markuplint> error: 属性「}」は許可されていません (invalid-attr) /home/xxx/sandbox/markuplint-issue-repo/src/App.tsx:12:10
  11: ••••••••••setValue(e.target.value•as•Value);
  12: ••••••••}}
  13: ••••••>
```

## The expected behavior

No Error

**Comment by daku10** [comment]
Thank you for a quick fix!
